### PR TITLE
First pass at using hostname

### PR DESCRIPTION
**Pull request by nhazekam**
Due to difference in internal/external Stampede IPs we want to allow for using the hostname that can be resolved correctly from either location. @btovar Please take a look, particularly in Makeflow where I pass as I'm not positive this is the cleanest way to pass this.

### CODE REVIEW

File: makeflow/src/makeflow.c [hunk 1]
```diff
@@ -1578,6 +1584,7 @@ int main(int argc, char *argv[])
 	batch_queue_set_option(remote_queue, "caching", cache_mode ? "yes" : "no");
 	batch_queue_set_option(remote_queue, "wait-queue-size", wq_wait_queue_size);
 	batch_queue_set_option(remote_queue, "working-dir", working_dir);
+	batch_queue_set_option(remote_queue, "master-prefer-hostname", master_prefer_hostname);
```
Comment (batrick) [inline makeflow/src/makeflow.c]: Make `master_prefer_hostname` an `int`. Set the the option to "yes" or "no" like we do for "caching" above.

File: work_queue/src/work_queue_worker.c [hunk 2]
```diff
@@ -1796,7 +1797,8 @@ static int serve_master_by_name( const char *catalog_host, int catalog_port, con
 		last_port = port;
 
 		debug(D_WQ,"selected master with project=%s name=%s addr=%s port=%d",project,name,addr,port);
-
+		if(pref && pref == 1)
```
Comment (batrick) [inline work_queue/src/work_queue_worker.c]: `if (pref)` ?

File: makeflow/src/makeflow.c (hunk 1 above)
Comment (nhazekam) [inline makeflow/src/makeflow.c]: Do you suggest I then pass yes or no on to the Catalog or equate those to values (1 or 0)  in batch_job_work_queue.c?

**Comment by batrick** [comment]
> Do you suggest I then pass yes or no on to the Catalog or equate those to values (1 or 0) in batch_job_work_queue.c?

You should do what's consistent with the rest of the code. [Do we use yes/no true/false 1/0 in boolean catalog fields?]

**Comment by btovar** [comment]
Nick,
Looks good. Please also update wavefront, allpairs, the respective show_help's, and the man pages.

**Comment by btovar** [comment]
Nick, did you also update the help and manual of makeflow?

**Comment by nhazekam** [comment]
I updated the help message and man page for makeflow, allpairs, and wavefront. Is there anything I need to add for work queue?

**Comment by btovar** [comment]
Nick,
A small change. IP and HOSTNAME give the idea that the user can write an IP or HOSTNAME.

Perhaps by_ip, by_hostname is better?

**Comment by btovar** [comment]
I've just tried on stampede, and it works.

**Comment by nhazekam** [comment]
@btovar Let me know how this looks. I tested on Stampede, but more testing never hurts.

**Comment by btovar** [comment]
Looks good. Ready to merge if you are.

**Comment by nhazekam** [comment]
I think its good to go.

**Pull request merged**